App.js

```diff
@@ -1,9 +1,363 @@
+      }
+      if (workEven === false && prevGameState.odd === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'ODD selected';
+      }
+      if (prevGameState.odd === false && workEven === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'EVEN selected';
+      }
+      return {
+        roll: prevGameState.roll,
+        even: workEven,
+        odd: prevGameState.odd,
+        position: prevGameState.position,
+        message: workMessage,
+        optMessage: workOptMessage,
+        score: prevGameState.score,
+        board: prevGameState.board,
+      };
+    });
+  }, []);
+
+  // press Odd button
+  const pressOdd = useCallback(() => {
+    setGameState(prevGameState => {
+      let workOdd;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if (prevGameState.odd === true) {
+        workOdd = false;
+      } else {
+        workOdd = true;
+      }
+      if (workOdd === true && prevGameState.even === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'cannot select both ODD and EVEN';
+        workOdd = false;
+      }
+      if (workOdd === true && prevGameState.even === false) {
+        workMessage = 'Roll again';
+        workOptMessage = 'ODD selected';
+      }
+      if (workOdd === false && prevGameState.even === true) {
+        workMessage = 'Roll again';
+        workOptMessage = 'EVEN selected';
+      }
+      return {
+        roll: prevGameState.roll,
+        even: prevGameState.even,
+        odd: workOdd,
+        position: prevGameState.position,
+        message: workMessage,
+        optMessage: workOptMessage,
+        score: prevGameState.score,
+        board: prevGameState.board,
+      };
+    });
+  }, []);
+
+  // press Roll button
+  const pressRoll = useCallback(() => {
+    setGameState(prevGameState => {
+      let workBoard = prevGameState.board.slice();
+      let randomNumber = Math.floor(Math.random() * 6) + 1;
+      let scoreAdj = 1;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if ((randomNumber === 1 || randomNumber === 3 || randomNumber === 5) && prevGameState.odd === true) {
+        randomNumber = randomNumber * 2;
+        workMessage = 'roll doubled';
+      }
+      if ((randomNumber === 2 || randomNumber === 4 || randomNumber === 6) && prevGameState.even === true) {
+        randomNumber = randomNumber * 2;
+        workMessage = 'roll doubled';
+      }
+      if ((randomNumber === 1 || randomNumber === 3 || randomNumber === 5) && prevGameState.even === true) {
+        randomNumber = 0;
+        workMessage = 'roll set to 0';
+      }
+      if ((randomNumber === 2 || randomNumber === 4 || randomNumber === 6) && prevGameState.odd === true) {
+        randomNumber = 0;
+        workMessage = 'roll set to 0';
+      }
+      // calculate location in array that matches current boardNumber
+      let filteredBoard = workBoard.filter(function (currentElement) {
```

Comment:
oh I like the use of `filter`!
I would change the inner function to an arrow function tho

Reply:
Rather then having to have a key field on every element of your board you can insert the key here before the filter since this is the inly spot you should be using it. Before the filter do a map to insert the key's and then do the filter after.

Reply:
But I reference key a few times within the render function called by flatlist.

Reply:
> oh I like the use of `filter`!
> I would change the inner function to an arrow function tho

Huh?

Reply:
So you switched it to generate it's own key and I chnaged to board's array to exclude the nkey.....and it started spitting out warnings saying I needed a key...so I switched it back.